2024-05-31-notes.md

```diff
@@ -0,0 +1,189 @@
+# 2024-05-31 - Jess and Orion notes
+
+Things we need to handle in version 2.0:
+
+- base properties for nodes
+- embed attached files
+
+- "extensions" through schemas
+
+- the schema doesn't have a single high-level version number. rather, each of the nodes and relations is defined by a schema with a version
+  - every node has a schema associated with it
+  - the base spec "includes" all of the node types in `@opencanvas/*`
+  - all of the schemas are currently embedded in file itself
+  - `x` and `y` are required when implementing a node type, optionally `z`
+
+Node Base class
+
+```ts
+type Node = {
+  id: string;
+  schema: string;
```

Comment:
`schema` and `schemaVersion` should be optional with a default for the most basic form of a node to keep simple things simple and to maintain some backwards compatibility with JSON Canvas 1.0.

Reply:
We're probably going to need to move `schema` and `schemaVersion` into the property bag and have (optionally) multiple property bags per node. See discussion [here](https://github.com/orgs/ocwg/discussions/19).